### ISSUE TITLE
OCSP responses with incorrect CertID don't cause ocsp tool to exit with error return code

### ISSUE THREAD

**Issue by tomato42** [openssl/openssl#7151]
<details><summary>ca.pem</summary><p>

```
-----BEGIN CERTIFICATE-----
MIIDJzCCAg+gAwIBAgIBATANBgkqhkiG9w0BAQsFADAVMRMwEQYDVQQKDApFeGFt
cGxlIENBMB4XDTEzMDkwNzE3MDcwM1oXDTI4MDkwNzE3MDcwM1owFTETMBEGA1UE
CgwKRXhhbXBsZSBDQTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALD1
Dto3INSfM/0pMzhBGd0xD+3szd5qo+9Q40WX6EJlNd+qmMnf9x0LKNUumTkrF589
Iyf4Oj4GZ65W43bVgch+2nEoXDdGuynRQwqOK+gPz8x5o2eDsPci+RYXWWgcHXpJ
Vzc6Rnej50FievlYueI22CQoAO85qmPqhcDyl6jjQ4mv5ysheAHiHwWOdOsomEEh
NhfZIJIEGpTpV/ocraRLDuGPL/KiYlexzJdFAoT4ftVAplOAGpYJPR6eswCs0jqs
cpJYIC/sd6sTkkShovBnQFMom4CknKqIvYUy1y2BNcckpPrO+sEMEtOPJasWKObJ
8uTivw3q669OCAtKGeECAwEAAaOBgTB/MA8GA1UdEwEB/wQFMAMBAf8wDgYDVR0P
AQH/BAQDAgEGMB0GA1UdDgQWBBTb4PUSHDpoMGb4hM/a0bWLhD2SkDA9BgNVHSME
NjA0gBTb4PUSHDpoMGb4hM/a0bWLhD2SkKEZpBcwFTETMBEGA1UECgwKRXhhbXBs
ZSBDQYIBATANBgkqhkiG9w0BAQsFAAOCAQEAWHu4S/0dCECwcC73jARKa3d1Bvpl
mCcE7OrwxJahbg+RG0L+jfufZVLr7P4wKkw1cMsQ0jae+DmbDPZaO6ELg4MYEDGI
vVzpdjp1Z7PxFb1heLj8WvLwx6b/Qyk4TdG4FcfOe8ca7vwy8dn6VQw8Ou4xZ5XF
lAu8GeJLC31SGPPQszCl7uT6q32WPTmQsy/4Yk+RFXNbgDTnkLzzIUDOz0nEcMft
rpYGLQYGcMLUfTZ+6D0LBn2/cT+/L+Dy4xXPyDXgPu+SzDrSCE1i99qemmDPw1Hx
a2k0lO17iie1dntLLgijLqDtrmj+Vir3hyMb8BxaE/AozWyeyw2b4BmO7g==
-----END CERTIFICATE-----
```

   </p></details>

  <details><summary>server.pem</summary><p>

```
-----BEGIN CERTIFICATE-----
MIIDXzCCAkegAwIBAgIBAjANBgkqhkiG9w0BAQsFADAVMRMwEQYDVQQKDApFeGFt
cGxlIENBMB4XDTE4MDkwNzE3MDcwM1oXDTE5MDkwNzE3MDcwM1owFDESMBAGA1UE
AwwJbG9jYWxob3N0MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAybTe
KO/l6/nqfCLW2g3DXPipA2zzFRKDJpj9z8BUU4gRATIBgeAP8Uv16yZOpuxtRQy5
9udAwzqoQWWpyA16jjBjsUXxU8D2+WQaFWaEudFgY1dGoh3PIHinkV2UurZ8a5qZ
zZXPNDpBN6jbR7M6dn6YRuUeLo5zrDdOCG9udqOJwgQmkWAOTKUO+pL4ZbWeZnfs
3YC9kWyzrC7Nih4b+G3iNxRprqy69OxQIPjpSMnkjBJiag3qOTomc+o3s9+aLucw
cVDdZc9/5WpODtKPgFRw3Th6ZEsvoIZ/wAVdTL5Ul6WaTMkOrxowmtGAtuXpoQZp
goRxquQxKE30DMEVUwIDAQABo4G6MIG3MA4GA1UdDwEB/wQEAwIDqDATBgNVHSUE
DDAKBggrBgEFBQcDATAdBgNVHQ4EFgQUdV8X1Gzs8jLX/FTsIUrONEFqFKwwPQYD
VR0jBDYwNIAU2+D1Ehw6aDBm+ITP2tG1i4Q9kpChGaQXMBUxEzARBgNVBAoMCkV4
YW1wbGUgQ0GCAQEwMgYIKwYBBQUHAQEEJjAkMCIGCCsGAQUFBzABhhZodHRwOi8v
bG9jYWxob3N0OjQyNDIvMA0GCSqGSIb3DQEBCwUAA4IBAQBYn7bAAJuzWTFUVWdM
XzfQ6MOeAVDItKgTp/CieU4O1e8IkH8GupqCfBTBOlrX1Pa4IDKN6jwSQCAztBPw
eOMdIiRqOBR46rMbZ91JHP42MKNnSiwpnuFYLCmZl7zuA9V/8ORmwA7j2e3ZceLN
8Vlm/e5HRLfnHUC5IIhBpvcTSNnDSlOEBa/9Zr86f1px6CyZ2Csmlhq/wbpeEbR9
qEu/gxboPxkSvZjayW52nwgBhvkPJK2mNYwZBW89J594IHdnjno1NMR4l3/wiwWD
7J7mQeg3DtFNpB9MfyNqfqqPSpwGL2bCyPNrXHebD9YXX/OMAubPgJ1ifCQ98PCw
FcxC
-----END CERTIFICATE-----
```

  </p></details>

  <details><summary>base64 of resp.pem</summary><p>

```
MIIBxgoBAKCCAb8wggG7BgkrBgEFBQcwAQEEggGsMIIBqDCBkaIWBBTb4PUSHDpoMGb4hM/a0bWL
hD2SkBgPMjAxODA5MDcxNzE1NTFaMGYwZDA8MAsGCWCGSAFlAwQCAwQUkBUkEUsQTCuZ8rp/mrdl
ULbfyK4EFNvg9RIcOmgwZviEz9rRtYuEPZKQAgECgAAYDzIwMTgwOTA3MTcxNTUxWqARGA8yMDE4
MDkwODE3MTU1MVowDQYJKoZIhvcNAQEFBQADggEBAK78pDygGmIDeRmp3sAeyTKTMgl1iXp6BuIp
QAl0ZwrmWwnXQvlCiMnOtZeGTaft1S9zGcto4PmaVnTUNTehDGFz6vTd0EtOriT4iSafpFcv3wSC
APEPIt5xKEn2ou0Mn5zXsl4SxzDxSsMyKsTm/HvcGB9q3AgwvX2iOQ8oJVPbXiAWVm1yb34ixwJM
RAPMKN5s2u2lhWbuBKljvHe0T+tffk4TAjpolk/idCvtdu/r/N3eguhVAcogTlAulz6R8kwOthjP
nO3wyraHFFWOq05sJxPVJskENDMs/qqnpOQaSLwsQIuGRxeYd2aOMdEp2FcHuwHyZCnjJpkr57Wv
kr4=
```

  </p></details>

Running the following command does not cause the the `ocsp` tool to report failure:
```
openssl ocsp -CAfile ca.pem -issuer ca.pem -verify_other ca.pem \
-cert server.pem -no_nonce -respin resp.pem
echo $?
```
Produces
```
Response verify OK
server.pem: ERROR: No Status found.
0
```

That's with 328a0547ad61d9e2

OpenSSL 1.0.2 produced:
```
Response Verify Failure
140099699771280:error:2706D077:OCSP routines:OCSP_MATCH_ISSUERID:unknown message digest:ocsp_vfy.c:319:
server/cert.pem: ERROR: No Status found.
1
```

**Comment by t8m** [comment]
Any chance to look at it?

**Comment by tomato42** [comment]
@openssl ping?

**Comment by mattcaswell** [comment]
Firstly I seem to get different output to you from 1.0.2:

````
Response Verify Failure
140275987326656:error:27069070:OCSP routines:OCSP_basic_verify:root ca not trusted:ocsp_vfy.c:168:
servertmp.pem: ERROR: No Status found.
````
Note the error message "ca not trusted". But you got "unknown message digest". Not sure why this is.

The difference in behaviour between 1.1x and 1.0.2 seems to be by design and as a result of (c6724060e):

https://rt.openssl.org/Ticket/Display.html?id=2206

That original RT ticket says:

> The attached patches (generated against OpenSSL 0.9.8n and OpenSSL-1.0.0-
> beta5) cause "openssl ocsp" to implicitly trust the Issuing CA Certificate (as 
> denoted by the "-issuer" parameter) as a candidate OCSP Response signer. This 
> "non-delegated" model is allowed by RFC 2560.

And goes on to say:

> But without this patch, you have to also specify "-VAfile ComodoEVSGCCA.crt" 
> to achieve the same result.

So if I amend the 1.0.2 command line to add "-VAfile ca.pem" I get the same result as in 1.1.x, i.e.

````
Response verify OK
server.pem: ERROR: No Status found.
0
````

**Comment by tomato42** [comment]
ok, that's weird, but I can't reproduce the `unknown message digest` error either...

that being said, the issue is as in the subject: that a failed verification attepmpt does not cause the ocsp tool to return with an error code

**Comment by tomato42** [comment]
verified fixed with 0ce47b35c7, thanks @t8m
